modules/lang-painless/spi/src/main/java/org/elasticsearch/painless/spi/annotation/PainlessAnnotation.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless.spi.annotation;
+
+public interface PainlessAnnotation {
```

Comment:
I'm not sure we need a base class. There are 2 cases we might want to use a name. The first is already the key of the map we pass into the whitelist parser. The second is if we want to add these annotations to the documentation of the whitelist. There I think we can use toString()? Even if we do eventually need a base class, let's add that when we have a concrete use for it.

Reply:
Done.